server/sonar-process/src/main/java/org/sonar/process/ProcessId.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.process;
+
+public enum ProcessId {
```

Comment:
This enum is a nice idea.
It would be neat to have it used everywhere.
When you rebased this branch on master, I don't think you updated master's code to use it (eg. https://github.com/SonarSource/sonarqube/blob/master/server/sonar-ce/src/main/java/org/sonar/ce/app/WebServerWatcherImpl.java#L42)